src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -241,11 +241,10 @@ export default class QueryEditorUI extends Component<Props, State> {
 
     if (!isEqual(this.props.query.dataset, prevQuery.dataset)) {
       if (this.inputRef) {
-        const newQuery = this.queryString.getInitialQuery();
-        const newQueryString = newQuery.query;
-        if (this.inputRef.getValue() !== newQueryString) {
+        const newQueryString = this.props.query.query;
+        if (this.inputRef.getValue() !== newQueryString && typeof newQueryString === 'string') {
```

Comment:
Why the type check?

Reply:
this.props.query.query can either be a 
` string | {
    [key: string]: any;
}`

So we only wanted to update the queryString. hence the type check